### PR TITLE
feat: allow use of gherlint config file

### PR DESCRIPTION
**Pull request by saw-jan**
## Description
The PR allows the use of `.gherlintrc(.js|.json)?` config file to customize the default gherlint behavior.

Current config:
```js
{
    files: [], // list of paths to look for feature files
    rules: {
        indentation: ["warn", 2]
    },
}
```
The following config files are accepted (priority in order):
```
1. .gherlintrc
2. .gherlintrc.js
3. .gherlintrc.json
```

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Tests only (no source changes)
- [ ] Documentation only (no source changes)

## Checklist:
- [x] Code changes
- [x] Unit tests added
- [ ] Documentation updated

### CODE REVIEW

File: lib/gherlint/GherlintConfig.js [hunk 1]
```diff
@@ -13,72 +13,80 @@ const Fs = require("../../utils/Fs");
...
+        //         log.error(
+        //             "'-c, --config' option takes file only.",
+        //             "Usage:",
+        //             "gherlint -c path/to/.gherlintrc"
+        //         );
+        //     }
+        //     return this.cliOptions.config;
+        // } else {
+        //     return this.#searchConfigFile();
+        // }
```
Comment (grgprarup) [inline lib/gherlint/GherlintConfig.js:84]: Do we need this?? or better to remove in this PR

File: lib/gherlint/GherlintConfig.js [hunk 2]
```diff
@@ -13,72 +13,80 @@ const Fs = require("../../utils/Fs");
...
+    }
 
-            return this.cliOptions.config;
-        } else {
-            return this.#searchConfigFile();
-        }
+    // Todo: implement --config cli option
+    // #hasCliConfigFileOption() {
+    //     return this.cliOptions.config ? true : false;
+    // }
```
Comment (grgprarup) [inline lib/gherlint/GherlintConfig.js:67]: Do we need this?

**Review by grgprarup — APPROVED** [review]
LGTM, other than existing ones.

File: lib/gherlint/GherlintConfig.js (hunk 2 above)
Comment (saw-jan) [reply lib/gherlint/GherlintConfig.js:67]: those are ancient code which I disabled with new structure/architecture. We want to implement them soon

**Pull request merged**